spec/jobs/update_s3_inventory_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+require "file_store/s3_store"
+
+describe Jobs::UpdateS3Inventory do
+  before do
+    SiteSetting.enable_s3_uploads = true
+    SiteSetting.s3_access_key_id = "abc"
+    SiteSetting.s3_secret_access_key = "def"
+    SiteSetting.enable_s3_inventory = true
+
+    store = FileStore::S3Store.new
+    @client = Aws::S3::Client.new(stub_responses: true)
+    store.s3_helper.stubs(:s3_client).returns(@client)
+    Discourse.stubs(:store).returns(store)
+  end
+
+  it "picks gravatar if system avatar is picked and gravatar was just downloaded" do
+    @client.expects(:put_bucket_policy)
+    @client.expects(:put_bucket_policy).with(
+      bucket: "bucket",
+      policy: "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Sid\":\"InventoryAndAnalyticsPolicy\",\"Effect\":\"Allow\",\"Principal\":{\"Service\":\"s3.amazonaws.com\"},\"Action\":[\"s3:PutObject\"],\"Resource\":[\"arn:aws:s3:::bucket/inventory/uploads/*\"],\"Condition\":{\"ArnLike\":{\"aws:SourceArn\":\"arn:aws:s3:::bucket\"},\"StringEquals\":{\"s3:x-amz-acl\":\"bucket-owner-full-control\"}}}]}"
+    )
+    @client.expects(:put_bucket_inventory_configuration)
```

Comment:
Is this line duplicated?

Reply:
No, it will be called twice for uploads and optimized images. Since both are identical I'm just checking the one (uploads) with the parameters.